src/rgw/rgw_admin.cc

```diff
@@ -2161,7 +2161,7 @@ int main(int argc, char **argv)
         return EINVAL;
       }
     } else if (ceph_argparse_witharg(args, i, &val, "--max-size", (char*)NULL)) {
-      max_size = strict_si_cast<int64_t>(val.c_str(), &err);
+      max_size = strict_si_cast<long long>(val.c_str(), &err);
```

Comment:
since the type of of `int64_t` is implementation defined, in other words, it's up to `cstdint` to make it an alias of `long int` or `long long`. maybe we can leave it as it is without the assumption? i know, technically `long long` is always 64bit long. but imho, it would be better use the type of the `max_size`. and be standard compliant. (c++ standard requires `long long` to be at least 64bit long)